python/pyspark/sql/connect/dataframe.py

```diff
@@ -139,11 +139,9 @@ def columns(self) -> List[str]:
         if self._plan is None:
             return []
         if "columns" not in self._cache and self._plan is not None:
-            pdd = self.limit(0).toPandas()
```

Comment:
maybe we don't need to cache `columns`, just return `self.schema.names`?

Reply:
I would prefer to not depend on the underly API when doing caching...

E.g. what if someday the cache on the schema is gone but this API is not aware of it, etc.

Reply:
Hmmm .. why do we need `_cache`? I think we can just remove.

Reply:
hmm if users call this API multiple times, we only need one gRPC. This should be useful right?

Reply:
Ok if you think this is a bit over-engineering I can remove.

Reply:
For that case, we should probably have a proper cache layer instead of doing this alone in `names`.

Reply:
we can do that for all cases for metadata-ish cases. e.g., schema or even collected results

Reply:
I removed this caching stuff in this PR. After we pretty much support enough API, we can go back to build a cache layer for all metadata like API to save RPC calls.